pkg/network/modules.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019 Red Hat and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modules define network function types
+
+// Package network contains all rendered network oriented resources for Nimbess
+package network
+
+import (
+	log "github.com/sirupsen/logrus"
+)
+
+// Gate is a virtual port for a module for traffic to ingress or egress
+type Gate uint64
+
+// PipelineModule type to implement module updates
+type PipelineModule interface {
+	Connect(module PipelineModule, egress bool, gate *Gate) Gate
+	Disconnect(gate Gate, egress bool)
+	GetNextIGate() Gate
+	GetNextEGate() Gate
+	GetName() string
+	SetName(s string)
+	GetIGateMap() map[Gate]PipelineModule
+	SetIGateMap(m map[Gate]PipelineModule)
+	GetEGateMap() map[Gate]PipelineModule
+	SetEGateMap(m map[Gate]PipelineModule)
+	SetMeta(s string)
+	GetMeta() string
+}
+
+// Module represents a generic Network Function.
+// It contains ingress gates and egress gates for traffic flow, which map to the next connected module.
+// Meta used to the name of the Meta Pipeline Module from which this module was created
```

Comment:
"used to the" doesn't sound right - is something missing?

Reply:
should be "Meta field is used to store the name of the Meta Pipeline Module from which this module was created". Will fix.